pandas/core/series.py

```diff
@@ -4726,6 +4726,13 @@ def rename(
                 errors=errors,
             )
         else:
+            # since _set_name supports only inplace and not copy param
+            # we determine inplace based on copy (if inplace is False)
+            # 1) If inplace is True, the value of copy is ignored
+            # 2) If inplace is False/not supplied(default case), then
+            #    inplace is determined based on copy (opposite of copy)
+            if not inplace:
+                inplace = not copy
             return self._set_name(index, inplace=inplace)
```

Comment:
the gist of the issue is that `_set_name` ( in the else portion ) doesn't support `copy` param and solely relies on `inplace` param.
Hence the `copy` param supplied to rename is just ignored in this case.

Some possible solutions here would be -
1. In the else portion, determine `inplace` based on `copy` ( in the case where `inplace` is False ) and then invoke `_set_name`
2. Support `copy` and `inplace` in `_set_name` and change its implementation similar to `rename`

I guess approach 1 might work/be better ( subject to review )

Reply:
hi @mrocklin could you please review this PR if possible or add the relevant reviewers for this ( since i'm not aware of the current owners of this part )

Thanks in advance,
Daniel.